concent_api/core/models.py

```diff
@@ -494,6 +494,70 @@ def clean(self) -> None:
+        if deserialized_report_computed_task.task_to_compute != deserialized_task_to_compute:
+            raise ValidationError({
+                'report_computed_task': 'Nested ReportComputedTask with TaskToCompute message '
+                                        'must be the same as both messages stored separately'
+            })
+        if self.ack_report_computed_task is not None:
+            if (
+                deserialize_message(self.ack_report_computed_task.data).report_computed_task !=
+                deserialized_report_computed_task
+            ):
+                raise ValidationError({
+                    'ack_report_computed_task': 'Nested ReportComputedTask with TaskToCompute message '
+                                                'must be the same as both messages stored separately'
+                })
+        if self.reject_report_computed_task is not None:
+            deserialized_reject_report_computed_task = deserialize_message(self.reject_report_computed_task.data)
+            if deserialized_reject_report_computed_task.reason == message.tasks.RejectReportComputedTask.REASON.GotMessageCannotComputeTask.name:  # pylint: disable=no-member
+                if deserialized_reject_report_computed_task.cannot_compute_task.task_to_compute != deserialized_task_to_compute:
+                    raise ValidationError({
+                        'reject_report_computed_task': 'Nested TaskToCompute in CannotComputeTask message must be '
+                                                       'the same as TaskToCompute which is stored separately'
+                    })
+            elif deserialized_reject_report_computed_task.reason == message.tasks.RejectReportComputedTask.REASON.GotMessageTaskFailure.name:  # pylint: disable=no-member
+                if deserialized_reject_report_computed_task.task_failure.task_to_compute != deserialized_task_to_compute:
+                    raise ValidationError({
+                        'reject_report_computed_task': 'Nested TaskToCompute in TaskFailure message must be '
+                                                       'the same as TaskToCompute which is stored separately'
+                    })
+            elif deserialized_reject_report_computed_task.reason == message.tasks.RejectReportComputedTask.REASON.SubtaskTimeLimitExceeded.name:  # pylint: disable=no-member
+                if deserialized_reject_report_computed_task.attached_task_to_compute != deserialized_task_to_compute:
+                    raise ValidationError({
+                        'reject_report_computed_task': 'Nested TaskToCompute message attached to RejectReportComputedTask '
+                                                       'must be the same as TaskToCompute which is stored separately'
+                    })
+            else:
+                raise ValidationError({
+                    'reject_repoert_computed_task': 'There is no REASON field in RejectReportComputedTask message.'
```

Comment:
actually, there must have been some reason, otherwise you would get `AttributeError` in line `515`. So the message should be more like:
```
f"Unknown reason: {deserialized_reject_report_computed_task.reason} for RejectReportComputedTask."
```

Reply:
BTW: reject_repoert_computed_task -> reject_**report**_computed_task

Reply:
Done and done